applications/audio/broadcast_configuration_tool/src/main.c

```diff
@@ -355,15 +332,98 @@ int main(void)
 	return 0;
 }
 
-static struct bt_bap_lc3_preset *preset_find(const char *name)
+static void context_list(const struct shell *shell)
 {
-	for (size_t i = 0; i < ARRAY_SIZE(bap_presets); i++) {
-		if (strcmp(bap_presets[i].name, name) == 0) {
-			return bap_presets[i].preset;
-		}
+	for (size_t i = 0; i < ARRAY_SIZE(contexts); i++) {
+		shell_print(shell, "%s", contexts[i].name);
 	}
+}
 
-	return NULL;
+static void broadcast_config_print(const struct shell *shell,
+				   struct broadcast_source_big *brdcst_param)
+{
+	int ret;
+
+	shell_print(shell, "\tPacking: %s",
+		    (brdcst_param->packing == BT_ISO_PACKING_INTERLEAVED ? "interleaved"
+									 : "sequential"));
+	shell_print(shell, "\tEncryption: %s",
+		    (brdcst_param->encryption == true ? "true" : "false"));
+
+	for (size_t i = 0; i < brdcst_param->num_subgroups; i++) {
+		struct bt_audio_codec_cfg *codec_cfg =
+			&brdcst_param->subgroups[i].group_lc3_preset.codec_cfg;
+
+		shell_print(shell, "\tSubgroup %d:", i);
+
+		shell_print(shell, "\t\tPreset: %s", brdcst_param->subgroups[i].preset_name);
+
+		int freq_hz = 0;
+
+		ret = le_audio_freq_hz_get(codec_cfg, &freq_hz);
+		if (ret) {
+			shell_error(shell, "Failed to get sampling rate: %d", ret);
+		} else {
+			shell_print(shell, "\t\tSampling rate: %dHz", freq_hz);
+		}
+
+		uint32_t bitrate = 0;
+
+		ret = le_audio_bitrate_get(codec_cfg, &bitrate);
+		if (ret) {
+			shell_error(shell, "Failed to get bitrate: %d", ret);
+		} else {
+			shell_print(shell, "\t\tBitrate: %dbps", bitrate);
+		}
+
+		int frame_duration = 0;
+
+		ret = le_audio_duration_us_get(codec_cfg, &frame_duration);
+		if (ret) {
+			shell_error(shell, "Failed to get frame duration: %d", ret);
+		} else {
+			shell_print(shell, "\t\tFrame duration: %dus", frame_duration);
+		}
+
+		uint32_t octets_per_sdu = 0;
+
+		ret = le_audio_octets_per_frame_get(codec_cfg, &octets_per_sdu);
+		if (ret) {
+			shell_error(shell, "Failed to get octets per frame: %d", ret);
+		} else {
+			shell_print(shell, "\t\tOctets per frame: %d", octets_per_sdu);
+		}
+
+		int language = bt_audio_codec_cfg_meta_get_stream_lang(codec_cfg);
+
+		if (language > 0) {
+			char lang[4] = {"\0"};
+
+			memcpy(lang, &language, 3);
+
+			shell_print(shell, "\t\tLanguage: %s", lang);
+		} else {
+			shell_print(shell, "\t\tLanguage: not_set");
+		}
+
+		shell_print(shell, "\t\tContext(s):");
+
+		for (size_t j = 0U; j < 16; j++) {
```

Comment:
Nitpick which I am not happy with myself.
```suggestion
		for (size_t j = 0U; j < sizeof(uint16_t) * CHAR_BIT; j++) {
```

Reply:
I added a comment above this:
`/* Context container is a bit field with length 16 */`